packages/core/src/components/widgets/CheckboxesWidget.tsx

```diff
@@ -0,0 +1,68 @@
+import { WidgetProps } from "@rjsf/utils";
+import React from "react";
+
+function selectValue(value: any, selected: any[], all: any[]) {
+  const at = all.indexOf(value);
+  const updated = selected.slice(0, at).concat(value, selected.slice(at));
+  // As inserting values at predefined index positions doesn't work with empty
+  // arrays, we need to reorder the updated selection to match the initial order
+  return updated.sort((a, b) => Number(all.indexOf(a) > all.indexOf(b)));
+}
+
+function deselectValue(value: any, selected: any[]) {
+  return selected.filter((v) => v !== value);
+}
+
+function CheckboxesWidget<T = any, F = any>({
```

Comment:
Can you add JSDoc here as well... take a look at how the other, already converted widgets are documented as a guide, same for the helper functions above.